langchain/src/llms/googlevertexai.ts

```diff
@@ -0,0 +1,387 @@
+    const projectId = await this.auth.getProjectId();
+    const url = `https://${this.endpoint}/v1/projects/${projectId}/locations/${this.location}/publishers/google/models/${this.model}:predict`;
+    const method = "POST" as const;
+
+    const data = {
+      instances,
+      parameters: {
+        temperature: this.temperature,
+        maxOutputTokens: this.maxTokens,
+        topP: this.topP,
+        topK: this.topK,
+      },
+    };
+
+    const opts = {
+      url,
+      method,
+      data,
+    };
+
+    async function request() {
+      return await client.request(opts);
+    }
+
+    const response = await this.caller.callWithOptions(
+      { signal: options.signal },
+      request.bind(client)
+    );
+
+    return <Response>response;
+  }
+
+  async predict(
+    prompt: string,
+    options: this["ParsedCallOptions"]
+  ): Promise<Prediction> {
+    const response = await this._predict(
+      [this.formatInstance(prompt)],
+      options
+    );
+    return this.formatResponse(response);
+  }
+
+  abstract formatInstance(prompt: string): GoogleVertexAiLLMInstance;
+
+  formatResponse(response: Response): Prediction {
+    return response?.data?.predictions[0];
+  }
+}
+
+/**
+ * Enables calls to the Google Cloud's Vertex AI API to access
+ * Large Language Models.
+ *
+ * To use, you will need to have one of the following authentication
+ * methods in place:
+ * - You are logged into an account permitted to the Google Cloud project
+ *   using Vertex AI.
+ * - You are running this on a machine using a service account permitted to
+ *   the Google Cloud project using Vertex AI.
+ * - The `GOOGLE_APPLICATION_CREDENTIALS` environment variable is set to the
+ *   path of a credentials file for a service account permitted to the
+ *   Google Cloud project using Vertex AI.
+ */
+export class GoogleVertexAiLLM extends GoogleVertexAiBaseLLM {
+  model = "text-bison";
+
+  constructor(fields?: GoogleVertexAiLLMInput) {
+    super(fields ?? {});
+
+    this.model = fields?.model ?? this.model;
+  }
+
+  formatInstance(prompt: string): GoogleVertexAiLLMTextInstance {
+    return { content: prompt };
+  }
+
+  predictionToGeneration(prediction: TextPrediction): Generation {
+    return {
+      text: prediction.content,
+      generationInfo: prediction,
+    };
+  }
+}
+
+/**
+ * Enables calls to the Google Cloud's Vertex AI API to access
+ * Large Language Models in a chat-like fashion.
+ *
+ * To use, you will need to have one of the following authentication
+ * methods in place:
+ * - You are logged into an account permitted to the Google Cloud project
+ *   using Vertex AI.
+ * - You are running this on a machine using a service account permitted to
+ *   the Google Cloud project using Vertex AI.
+ * - The `GOOGLE_APPLICATION_CREDENTIALS` environment variable is set to the
+ *   path of a credentials file for a service account permitted to the
+ *   Google Cloud project using Vertex AI.
+ */
+export class GoogleVertexAiChat
```

Comment:
Can you move this under `chat_models`? E.g.:

langchain/src/chat_models/anthropic.ts

Reply:
Sure, tho I had it here since the OpenAI chat is also in LLM

Reply:
I guess, perhaps, I'm confused about the openai-chat.ts code and what it's there for, since I do see it under chat_models/openai.ts. Will move it there.

Reply:
The idea was to have an entrypoint for newer people who weren't aware `gpt-3.5-turbo` and `gpt-4` are a separate class of model:

https://github.com/hwchase17/langchainjs/blame/main/langchain/src/llms/openai.ts#L112

Definitely a bit confusing and something we should clarify better going forwards.